packages/manager/cypress/e2e/core/account/account-maintenance.spec.ts

```diff
@@ -118,12 +119,140 @@ describe('Maintenance', () => {
         });
       });
 
-    // Confirm download buttons work
-    cy.get('button')
-      .filter(':contains("Download CSV")')
-      .should('be.visible')
-      .should('be.enabled')
-      .click({ multiple: true });
-    // TODO Need to add assertions to confirm CSV contains the expected contents on first trial (M3-8393)
+    // Validate content of the downloaded CSV for pending maintenance
+    cy.get('a[download*="pending-maintenance"]')
+      .invoke('attr', 'download')
+      .then((fileName) => {
+        const downloadsFolder = Cypress.config('downloadsFolder');
+
+        // Delete the file before the test
+        cy.exec(`rm -f ${downloadsFolder}/${fileName}`).then((result) => {
+          if (result.code === 0) {
+            cy.log(`Deleted file: ${fileName}`);
+          } else {
+            cy.log(`Failed to delete file: ${fileName}`);
+          }
+        });
+
+        // Locate the <a> element for pending-maintenance and then find its sibling <button> element
+        cy.get('a[download*="pending-maintenance"]')
+          .siblings('button')
+          .should('be.visible')
+          .and('contain', 'Download CSV')
+          .click();
+
+        // Map the expected CSV content to match the structure of the downloaded CSV
+        const expectedPendingMigrationContent = accountpendingMaintenance.map(
+          (maintenance) => ({
+            entity_label: maintenance.entity.label,
+            entity_type: maintenance.entity.type,
+            type: maintenance.type,
+            status: maintenance.status,
+            reason: maintenance.reason,
+          })
+        );
+
+        // Read the downloaded CSV and compare its content to the expected CSV content
+        cy.readFile(`${downloadsFolder}/${fileName}`)
+          .then((csvContent) => {
+            const parsedCsvPendingMigration = parseCsv(csvContent);
```

Comment:
I'm noticing an intermittent failure here (and later in the spec when we download the completed maintenance CSV) where `csvContent` is apparently `''`. It seems like there's a brief moment where the CSV file exists but is empty, and an empty string gets passed to the `parseCsv` function which throws an error.

I read a bit into `cy.readFile` [here](https://docs.cypress.io/api/commands/readfile#Automatic-retries), and it looks like adding some assertions on the file contents allows Cypress to wait for the data to actually show up and seems to resolve the issue:

```suggestion
        cy.readFile(`${downloadsFolder}/${fileName}`)
          .should('not.eq', null)
          .should('not.eq', '')
          .then((csvContent) => {
            const parsedCsvPendingMigration = parseCsv(csvContent);
```

Wondering if you ran into this issue at all? My machine is really slow and this issue seems to be pretty time sensitive so I wonder if this occurs at all for developers with newer/faster machines.

Reply:
I'm on an M3 chip, wasn't running into any issues

Reply:
same for me, I didn't came across failure as of now on my machine. Thanks for pointing it out, I did suggested change in this commit https://github.com/linode/manager/pull/11168/commits/764fed390f63f679fd0abf96dcf9f13cb4e5f389